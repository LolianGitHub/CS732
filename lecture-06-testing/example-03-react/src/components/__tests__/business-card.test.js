import React from 'react';
import renderer from 'react-test-renderer';
import BusinessCard from '../business-card';

it('snapshot with no content matches', () => {
    const tree = renderer.create(<BusinessCard />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('snapshot with name matches', () => {
    const tree = renderer.create(<BusinessCard name="Bob" />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('snapshot with phone number matches', () => {
    const tree = renderer.create(<BusinessCard phNum="021 123 4567" />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('snapshot with email matches', () => {
    const tree = renderer.create(<BusinessCard email="bob@bob.com" />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('snapshot with all matches', () => {
    const tree = renderer.create(<BusinessCard name="Bob" phNum="021 123 4567" email="bob@bob.com" />).toJSON();
    expect(tree).toMatchSnapshot();
});