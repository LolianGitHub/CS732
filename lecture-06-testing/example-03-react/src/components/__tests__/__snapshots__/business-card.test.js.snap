// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot with all matches 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        Name:
      </th>
      <td>
        Bob
      </td>
    </tr>
    <tr>
      <th>
        Phone number:
      </th>
      <td>
        021 123 4567
      </td>
    </tr>
    <tr>
      <th>
        Email:
      </th>
      <td>
        bob@bob.com
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`snapshot with email matches 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        Email:
      </th>
      <td>
        bob@bob.com
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`snapshot with name matches 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        Name:
      </th>
      <td>
        Bob
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`snapshot with no content matches 1`] = `
<table>
  <tbody />
</table>
`;

exports[`snapshot with phone number matches 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        Phone number:
      </th>
      <td>
        021 123 4567
      </td>
    </tr>
  </tbody>
</table>
`;
